sdk/core/azure-core-test/CMakeLists.txt

```diff
@@ -45,5 +45,5 @@ target_include_directories (azure-core-test-fw
 
 # make sure that users can consume the project as a library.
 add_library (Azure::Core::Test ALIAS azure-core-test-fw)
-target_link_libraries(azure-core-test-fw PRIVATE azure-core gtest)
+target_link_libraries(azure-core-test-fw PRIVATE azure-core Azure::azure-identity gtest)
```

Comment:
Why does azure-core-test depend on azure-identity? This means we can't consume this project as a standalone, or exclude building azure-identity and still have core/core-tests work.

It looks like we introduced this for test_base.hpp adding `CreateClientSecretCredential`, which we then use in Attestation unit tests (and only there). I think this dependency should move there and removed from `test_base.hpp` in azure-core. OR, we find some other mechanism of having a "core" test, but one that doesn't build with azure-core. 

Having azure-core-test depend on something upstream like azure-identity is inverted to me.